### PR TITLE
bpo-45057: Simplify RegressionTestResult

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
Remove code which duplicates the functionality of TextTestResult.

https://bugs.python.org/issue45057

### CODE REVIEW

File: Lib/test/support/testresult.py [hunk 1]
```diff
@@ -111,45 +101,26 @@ def __makeErrorDict(cls, err_type, err_value, err_tb):
...
-    def printErrors(self):
-        if self.__verbose:
-            self.stream.write('\n')
-        self.printErrorList('ERROR', self.errors)
-        self.printErrorList('FAIL', self.failures)
-
-    def printErrorList(self, flavor, errors):
-        for test, err in errors:
-            self.stream.write(self.separator1)
-            self.stream.write(f'{flavor}: {self.getDescription(test)}\n')
```
Comment (vstinner) [inline Lib/test/support/testresult.py:150]: You can modernize printErrors() in Lib/unittest/runner.py. It contains very useless code like `"%s" % err`.

**Review by vstinner — APPROVED** [review]
LGTM. Yeah, it looks like most code already exists in Lib/unittest/runner.py.

I didn't check manually if the output remains the same.

File: Lib/test/support/testresult.py (hunk 1 above)
Comment (serhiy-storchaka) [reply Lib/test/support/testresult.py:150]: Maybe in other PR. This PR does not touch that file.

**Comment by vstinner** [comment]
For me this change is just refactoring and not bugfix, you can backport it, but usually I prefer to leave stable branches unchanged.

File: Lib/test/support/testresult.py (hunk 1 above)
Comment (vstinner) [reply Lib/test/support/testresult.py:150]: As you want, it was just a remark. I approved your PR (and you merged it ;-))

**Comment by serhiy-storchaka** [comment]
It is internal code, and it can help when backport changes to TextTestResult. I prefer to backport test changes.

**Comment by vstinner** [comment]
> Hi! The buildbot x86-64 macOS 3.10 has failed when building commit 3d56272.

This is https://bugs.python.org/issue43592

**Pull request merged**